### PR TITLE
[CLI] Add show tables for datafusion-cli

### PR DESCRIPTION
**Pull request by gaojun2048**
closes https://github.com/apache/arrow-datafusion/issues/1984

### CODE REVIEW

File: datafusion-cli/src/main.rs [hunk 1]
```diff
@@ -82,6 +82,9 @@ struct Args {
         help = "Reduce printing other than the results and work quietly"
     )]
     quiet: bool,
+
+    #[clap(long, help = "whether enable with_information_schema")]
```
Comment (andygrove) [inline datafusion-cli/src/main.rs:86]:
```suggestion
    #[clap(long, help = "Enable with_information_schema")]
```

File: datafusion-cli/src/main.rs [hunk 2]
```diff
@@ -82,6 +82,9 @@ struct Args {
         help = "Reduce printing other than the results and work quietly"
     )]
     quiet: bool,
+
+    #[clap(long, help = "Enable with_information_schema")]
```
[the remark this answers is not in the document]
Comment (matthewmturner) [reply datafusion-cli/src/main.rs:86]: I agree with the above

[the remark this answers is not in the document]
Comment (gaojun2048) [reply datafusion-cli/src/main.rs:86]: I agree too!

File: datafusion-cli/src/context.rs [hunk 3]
```diff
@@ -59,7 +59,10 @@ impl BallistaContext {
     pub async fn try_new(host: &str, port: u16) -> Result<Self> {
         use ballista::context::BallistaContext;
         use ballista::prelude::BallistaConfig;
-        let config: BallistaConfig = BallistaConfig::new()
+        let builder =
```
Comment (alamb) [inline datafusion-cli/src/context.rs:62]: ❤️

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @gaojun2048  -- looks good to me

**Pull request merged**